lib/msf/core/db_manager/note/initialize_notes.rb

```diff
@@ -0,0 +1,120 @@
+
+module Msf::DBManager::Note::InitializeNotes
+  def initialize_note(opts)
+    return if not active
+    ::ApplicationRecord.connection_pool.with_connection {
+      wspace = Msf::Util::DBManager.process_opts_workspace(opts, framework)
+      opts = opts.clone()
+      opts.delete(:workspace)
+      seen = opts.delete(:seen) || false
+      crit = opts.delete(:critical) || false
+      host = nil
+      addr = nil
+      # Report the host so it's there for the Proc to use below
+      if opts[:host]
+        if opts[:host].kind_of? ::Mdm::Host
+          host = opts[:host]
+        else
+          addr = Msf::Util::Host.normalize_host(opts[:host])
+          host = report_host({:workspace => wspace, :host => addr})
+        end
+        # Do the same for a service if that's also included.
+        if (opts[:port])
+          proto = nil
+          sname = nil
+          proto_lower = opts[:proto].to_s.downcase # Catch incorrect usages
+          case proto_lower
+          when 'tcp','udp'
+            proto = proto_lower
+            sname = opts[:sname] if opts[:sname]
+            # XXX: These normalizations are lazy af
+          when 'http', 'smb'
+            proto = 'tcp'
+            sname = proto_lower
+          when 'dns','snmp','dhcp'
+            proto = 'udp'
+            sname = proto_lower
+          else
+            proto = 'tcp'
+            sname = proto_lower
+          end
+          sopts = {
+            :workspace => wspace,
+            :host  => host,
+            :port  => opts[:port],
+            :proto => proto
+          }
+          sopts[:name] = sname if sname
+          report_service(sopts)
+        end
+      end
+      # Update Modes can be :unique, :unique_data, :insert
+      mode = opts[:update] || :unique
+
+
+      if addr and not host
+        host = get_host(:workspace => wspace, :host => addr)
+      end
+      if host and (opts[:port] and proto)
```

Comment:
Were `and`/`not` keyword semantics specifically desired here? If not 
```suggestion
      if addr && !host
        host = get_host(:workspace => wspace, :host => addr)
      end
      if host && (opts[:port] && proto)
```

I suspect this code just moved but worth testing and updating.

Reply:
yup, you suspect right, just moved, will update though :+1: